pkg/tcpip/network/ipv4/icmp.go

```diff
@@ -133,6 +133,43 @@ func (e *endpoint) handleICMP(r *stack.Route, netHeader buffer.View, vv buffer.V
 
 	case header.ICMPv4Timestamp:
 		received.Timestamp.Increment()
+		// Only send a reply if the checksum is valid.
+         	wantChecksum := h.Checksum()
+        	// Reset the checksum field to 0 to can calculate the proper
+        	// checksum. We'll have to reset this before we hand the packet off.
+        	h.SetChecksum(0)
+            	gotChecksum := ^header.ChecksumVV(vv, 0 /* initial */)
+            	if gotChecksum != wantChecksum {
+                	// It's possible that a raw socket expects to receive this.
+                	h.SetChecksum(wantChecksum)
+                	e.dispatcher.DeliverTransportPacket(r, header.ICMPv4ProtocolNumber, netHeader, vv)
+                	received.Invalid.Increment()
+           	 	return
+        	}
+        	// It's possible that a raw socket expects to receive this.
+        	h.SetChecksum(wantChecksum)
+        	e.dispatcher.DeliverTransportPacket(r, header.ICMPv4ProtocolNumber, netHeader, vv)
+
+        	vv := vv.Clone(nil)
+        	vv.TrimFront(header.ICMPv4TimeStampMinimumSize)
+        	hdr := buffer.NewPrependable(int(r.MaxHeaderLength()) + header.ICMPv4TimeStampMinimumSize)
+        	pkt := header.ICMPv4(hdr.Prepend(header.ICMPv4TimeStampMinimumSize))
+        	copy(pkt, h)
+        	pkt.SetType(header.ICMPv4TimestampReply)
+
+        	now_tx := time.Now()
+       	 	time_tx_s := ((now_tx.UnixNano()/1000000) % header.MsInDay)
```

Comment:
Beyond the inline const suggestion, you could also do:

now := ...
lastDay := now.Truncate(24 * time.Hour)
ms := now.Since(lastDay).Milliseconds()

Actually I think this is clearer.